File: trunk/server/src/logic/prpc/EquipmentSlot.go
package prpc

// enum EquipmentSlot
const (
	ES_MIN      = 0
	ES_HEAD     = 1
	ES_BODY     = 2
	ES_FEET     = 3
	ES_ORNAMENT = 4
	ES_WEAPON   = 5
	ES_MAX      = 6
)
const (
	K_ES_MIN      = "ES_MIN"
	K_ES_HEAD     = "ES_HEAD"
	K_ES_BODY     = "ES_BODY"
	K_ES_FEET     = "ES_FEET"
	K_ES_ORNAMENT = "ES_ORNAMENT"
	K_ES_WEAPON   = "ES_WEAPON"
	K_ES_MAX      = "ES_MAX"
)

func ToName_EquipmentSlot(id int) string {
	switch id {
	case ES_MIN:
		return "ES_MIN"
	case ES_HEAD:
		return "ES_HEAD"
	case ES_BODY:
		return "ES_BODY"
	case ES_FEET:
		return "ES_FEET"
	case ES_ORNAMENT:
		return "ES_ORNAMENT"
	case ES_WEAPON:
		return "ES_WEAPON"
	case ES_MAX:
		return "ES_MAX"
	default:
		return ""
	}
}
func ToId_EquipmentSlot(name string) int {
	switch name {
	case "ES_MIN":
		return ES_MIN
	case "ES_HEAD":
		return ES_HEAD
	case "ES_BODY":
		return ES_BODY
	case "ES_FEET":
		return ES_FEET
	case "ES_ORNAMENT":
		return ES_ORNAMENT
	case "ES_WEAPON":
		return ES_WEAPON
	case "ES_MAX":
		return ES_MAX
	default:
		return -1
	}
}

File: trunk/Client/Assets/Scripts/Common/Rpc/EquipmentSlot.cs
public class EquipmentSlot{
  static string[] _NAMES = new string[]{
    "ES_MIN",
    "ES_HEAD",
    "ES_BODY",
    "ES_FEET",
    "ES_ORNAMENT",
    "ES_WEAPON",
    "ES_MAX"
  };
  public const int ES_MIN = 0;
  public const int ES_HEAD = 1;
  public const int ES_BODY = 2;
  public const int ES_FEET = 3;
  public const int ES_ORNAMENT = 4;
  public const int ES_WEAPON = 5;
  public const int ES_MAX = 6;
  public static int ToId(string name){
    for(int i=0; i<NAMES.Length;++i){
      if(_NAMES[i] == name){
        return i;
      }
    }
    return -1;
  }
  public static string ToName(int id){
    if(id<0||id>=_NAMES.Length){
      return "";
    }
    return _NAMES[id];
  }
  public static string[] NAMES{ get {return _NAMES;} }
}
